crates/bevy_render/src/render_resource/gpu_array_buffer.rs

```diff
@@ -0,0 +1,129 @@
+use super::StorageBuffer;
+use crate::{
+    render_resource::batched_uniform_buffer::BatchedUniformBuffer,
+    renderer::{RenderDevice, RenderQueue},
+};
+use bevy_ecs::{prelude::Component, system::Resource};
+use encase::{private::WriteInto, ShaderSize, ShaderType};
+use std::{marker::PhantomData, mem};
+use wgpu::{BindGroupLayoutEntry, BindingResource, BindingType, BufferBindingType, ShaderStages};
+
+/// Trait for types able to go in a [`GpuArrayBuffer`].
+pub trait GpuArrayBufferable: ShaderType + ShaderSize + WriteInto + Clone {}
+impl<T: ShaderType + ShaderSize + WriteInto + Clone> GpuArrayBufferable for T {}
+
+/// Stores a list of elements to be transferred to the GPU and made accessible to shaders as a read-only array.
```

Comment:
Just to keep naming a bit more consistent.

```suggestion
/// Stores an array of elements to be transferred to the GPU and made accessible to shaders as a read-only array.
```